src/pybind/mgr/dashboard/frontend/src/app/shared/components/cd-label/cd-label.component.ts

```diff
@@ -0,0 +1,47 @@
+import { AfterContentInit, Component, ElementRef, Input, ViewChild } from '@angular/core';
+
+import { FixedColorClass } from '~/app/shared/enum/fixed-color-class.enum';
+
+@Component({
+  selector: 'cd-label',
+  templateUrl: './cd-label.component.html',
+  styleUrls: ['./cd-label.component.scss']
+})
+export class CdLabelComponent implements AfterContentInit {
+  @ViewChild('contentWrapper')
+  content_child: ElementRef;
+
+  // element key, for key-value elements
+  @Input() key?: string;
+
+  element_text: string;
+  // number for the color class
+  color_class: string;
+
+  // number of colors of the color pool
+  readonly n_colors = 6;
+
+  ngAfterContentInit() {
+    if (this.key) {
+      if (this.key in FixedColorClass) {
+        this.color_class = FixedColorClass[this.key];
+      } else {
+        this.color_class = this.stringToClassColorNumber(this.key).toString();
+      }
+    } else if (this.content_child) {
+      this.element_text = this.content_child.nativeElement.innerText;
+      this.color_class = this.stringToClassColorNumber(this.element_text).toString();
+    }
+  }
+
+  stringToClassColorNumber(text: string): number {
+    let hash = 0;
+    let char_code = 0;
+    for (let i = 0; i < text.length; i++) {
+      char_code = text.charCodeAt(i);
+      // tslint:disable-next-line:no-bitwise
+      hash = (hash << 5) - hash + char_code;
+    }
+    return hash % this.n_colors;
+  }
```

Comment:
If you combine all this with a property getter, you wouldn't need to implement the ngAfterContentInit hook. Something like:
```ts
  get  color_class() {
    cssClasses['fa-' + this.icon] = true;
    return this.stringToClassColorNumber(this.element_text).toString();;
  }
```
However if we find that content projection becomes an issue, or results in complex code that has to handle advanced hooks, there are a couple of alternatives:
* Passing the label text as another input arg. That's pretty common in Angular components. In fact, component projection is mainly used when the component is made of nested HTML elements. In our, we expect it to be just text.
* Using a pipe like `<span class="badge">{{ text | customColor }}</span>`. But that will probably require [`innerHTML `](https://stackoverflow.com/questions/46955375/angular2-pipe-for-changing-inline-style-of-specific-character).

Reply:
Updated the component and the description of the PR @epuertat, I'll wait for your thoughts on the changes I made based on your comments, let me know what you think. 
Thank you!